atlasdb-config/src/main/java/com/palantir/atlasdb/config/PersistentStorageConfig.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(
+                value = ImmutableRocksDbPersistentStorageConfig.class,
+                name = RocksDbPersistentStorageConfig.TYPE)})
+public interface PersistentStorageConfig {
+    String type();
+
+    /**
+     * Path to the directory in which we want to store the data. In case the directory does not exist we create a new
```

Comment:
We should say this must be a relative path.

Reply:
Comment is now in line with the implementation.